packages/renderer/src/lib/recommendation/ExtensionBanner.spec.ts

```diff
@@ -60,8 +60,24 @@ const imageBackground: IExtensionBanner = {
   },
 };
 
+const showMessageBoxMock = vi.fn();
+const telemetryTrackMock = vi.fn();
+
+beforeAll(() => {
+  Object.defineProperty(window, 'showMessageBox', {
+    value: showMessageBoxMock,
+  });
+  Object.defineProperty(window, 'telemetryTrack', {
+    value: telemetryTrackMock,
+  });
+  Object.defineProperty(window, 'updateConfigurationValue', {
+    value: vi.fn(),
+  });
+});
+
 beforeEach(() => {
   vi.resetAllMocks();
+  telemetryTrackMock.mockImplementation(() => Promise.resolve());
```

Comment:
if you look at recommended pattern, it's always vi.mocked(window.telemetryTrack)

that said I don't know why we would need an empty implementation ?